assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js

```diff
@@ -39,7 +39,7 @@ import {
 
 function useUploadMedia({ media, pagingNum, mediaType, fetchMedia, setMedia }) {
   const {
-    state: { globalHistoryLength },
+    state: { versionNumber },
```

Comment:
I don't necessarily see that the uploading has much to do with history itself. I think we're trying to hack away here from a somewhat different underlying structure that involves a hook and a provider. I think trying to skip the provider, makes it harder for us here. Let's discuss something like this:

/1/ There's a high-level `PreventWindowUnloadProvider`. It it exposes the action:
```
const setPreventUnload = usePreventWindowUnloadHook();
```

/2/  The history work is all the same and it calls:
```
const setPreventUnload = usePreventWindowUnloadHook();
useEffect(() => {
  setPreventUnload('history', hasChangedSinceLastSave);
}, [hasChangedSinceLastSave]);
```

/3/ The uploader is about the same:
```
const setPreventUnload = usePreventWindowUnloadHook();
// We can either use a single "upload" ID, or a unique ID per uploading session.
setPreventUnload('upload', true);
uploadFile(...).finally(() => setPreventUnload('upload', false));
```

/4/ The `PreventWindowUnloadProvider` is thus pretty simple:
```
function PreventWindowUnloadProvider() {
  const lockRef = useRef(new Set());
  const setPreventUnload = useCallback((id, value) => {
    if (value) {
      lockRef.current.add(id);
    } else {
      lockRef.current.delete(id);
    }
  }, []);
  useEffect(() => {
    window.addEventListener(..., (evt) => {
      if (lockRef.current.size > 0) {
        evt.preventDefault();
        // ....
      }
    });
    return () => window.removeEventListener(...);
  }, []);
}
```